x-pack/filebeat/tests/integration/input_metrics_test.go

```diff
@@ -0,0 +1,315 @@
+    allow_deprecated_use: true
+
+  - type: filestream
+    id: {{.filestream_id}}
+    enabled: true
+    paths:
+      - {{.log_path}}
+    processors:
+      - drop_event:
+          when:
+            contains:
+              message: "PUT"
+    close.reader.after_interval: 10m
+
+  - type: cel
+    id: {{.cel_id}}
+    interval: 1s
+    resource.url: {{.cel_resource_url}}
+    program: bytes(get(state.url).Body).as(body,{"events":[body.decode_json()]})
+    publisher_pipeline.disable_host: true
+    processors:
+      - drop_event:
+          when:
+            equals:
+              ip: "1.1.1.1"
+
+  - type: httpjson
+    id: {{.httpjson_id}}
+    interval: 1s
+    request.url: {{.httpjson_requestURL}}
+    processors:
+      - drop_event:
+          when:
+            contains:
+              message: "1.1.1.1"
+
+queue.mem:
+  events: 32
+  flush.min_events: 8
+  flush.timeout: 0.1s
+
+path.home: {{.path_home}}
+
+output.file:
+  path: ${path.home}
+  filename: output-file
+  rotate_every_kb: 10000
+
+logging.level: debug
+`
+
+	celSrv := makeServer()
+	defer celSrv.Close()
+	httpjsonSrv := makeServer()
+	defer httpjsonSrv.Close()
+
+	filebeat := NewFilebeat(t)
+	tempDir := filebeat.TempDir()
+
+	// 1. Generate the log file path
+
+	relativePath := filepath.Join("testdata", "input_metrics.log")
+	logFilePath, err := filepath.Abs(relativePath)
+	require.NoError(t, err, "Failed to get absolute path for", relativePath)
+
+	relativePath = filepath.Join("testdata", "input_metrics-no-id.log")
+	logFileNoIDPath, err := filepath.Abs(relativePath)
+	require.NoError(t, err, "Failed to get absolute path for", relativePath)
+
+	// 2. Write configuration file and start Filebeat
+	cgfSB := strings.Builder{}
+	tmpl, err := template.New("filebeatConfig").Parse(tmplCfg)
+	require.NoErrorf(t, err, "Failed to parse config template")
+
+	filestreamInputID := "a-filestream-id"
+	celBaseInputID := "a-cel-input-id"
+	celInputID := fmt.Sprintf("%s::%s", celBaseInputID, celSrv.URL)
+	httpsjonInputID := "a-httpjson-input-id"
+
+	require.NoError(t, tmpl.Execute(&cgfSB, map[string]string{
+		"filestream_id":       filestreamInputID,
+		"cel_id":              celBaseInputID,
+		"httpjson_id":         httpsjonInputID,
+		"log_path":            logFilePath,
+		"log_path_no_id":      logFileNoIDPath,
+		"cel_resource_url":    celSrv.URL,
+		"httpjson_requestURL": httpjsonSrv.URL,
+		"path_home":           tempDir,
+	}), "failed to execute config template")
+
+	filebeat.WriteConfigFile(cgfSB.String())
+	filebeat.Start()
+
+	// 4. Wait for Filebeat to start scanning for files
+	filebeat.WaitForLogs(
+		fmt.Sprintf("A new file %s has been found", logFilePath),
+		30*time.Second,
+		"Filebeat did not start looking for files to ingest")
+
+	filebeat.WaitForLogs(
+		fmt.Sprintf("End of file reached: %s; Backoff now.", logFilePath),
+		10*time.Second, "Filebeat did not close the file")
+
+	// 5. Now that the reader has been closed, we can make the assertions.
```

Comment:
It's a detail, but `End of file reached` does not mean the reader has been closed, it only means the file was read until the end.

The reader closed log: https://github.com/elastic/beats/blob/1db044487726833ee5a63f98904326d705707c46/filebeat/input/filestream/input.go#L368